# Stationarity and Non-Stationarity

- $H_0$: Null Hypothesis assumes Non Stationarity. ie Autocorrelation coefficient is less than 1
$$\varphi < 1$$
- $H_1$: Alternative Hypothesis assumes Stationarity. ie Autocorrelation coefficient is 1
$$\varphi = 1$$
$$test\_statistic < critical\_value \geq stationarity$$ 

In [1]:
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.graphics.tsaplots as sgt
import statsmodels.tsa.stattools as sts
from statsmodels.tsa.seasonal import seasonal_decompose
import seaborn as sns
sns.set()
DATA_DIR = '../data'

In [2]:
import utils

In [3]:
df, df_test = utils.get_index_2018_market_value_splits()

/Users/shankar/dev/code/ds/main/data_science/coursera/u-time_series_analysis/s05_working_with_ts/utils.py:32: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['wn'] = wn


## Dickey Fuller Test

In [4]:
sts.adfuller(df.market_value)

(-1.7369847452352425,
 0.4121645696770626,
 18,
 5002,
 {'1%': -3.431658008603046,
  '5%': -2.862117998412982,
  '10%': -2.567077669247375},
 39904.880607487445)

- -1.7369847452352425, - `Test(t) Statistic`
- 0.4121645696770626, - `p-Value`, 40% chance that not rejecting the null
- 18, - `No of lags` used in regression while determining the statistics
- 5002, - `No of observations`
- {'1%': -3.431658008603046,
  '5%': -2.862117998412982,
  '10%': -2.567077669247375}, - `Critical Value`
- 39904.880607487445 - `auto correlation`

### Observations
- t-Statistics is greater than any of the critical values

## On White Noise

In [5]:
sts.adfuller(df.wn)

(-51.642990990789684,
 0.0,
 1,
 5019,
 {'1%': -3.4316535759402753,
  '5%': -2.8621160400844468,
  '10%': -2.567076626752987},
 70999.04550285518)

In [7]:
df['rw'] = utils.get_random_walk_data()
df.head()

market_value           wn           rw
date                                              
1994-01-07        469.90  1167.122767  1122.139662
1994-01-10        475.27   979.293474  1080.347860
1994-01-11        474.13  1210.826868  1082.095245
1994-01-12        474.17  1072.766098  1083.639265
1994-01-13        472.47  1479.942338  1067.146255

## On Random Walk

In [9]:
sts.adfuller(df.rw)

(-1.3286073927689726,
 0.6159849181617382,
 24,
 4996,
 {'1%': -3.4316595802782865,
  '5%': -2.8621186927706463,
  '10%': -2.567078038881065},
 46299.333497595144)

- Covariances of two such intervals with identical size would very rarely be equal
- We expect Random walks to be a non-stationary process